src/subdomains/supporting/support-issue/services/support-issue.service.ts

```diff
@@ -121,6 +115,38 @@ export class SupportIssueService {
     await this.messageRepo.save(entity);
 
     if (dto.author !== CustomerAuthor) await this.supportIssueNotificationService.newSupportMessage(entity);
+
+    return SupportIssueDtoMapper.mapSupportMessage(entity);
+  }
+
+  async getSupportIssue(userDataId: number, query: GetSupportIssueFilter): Promise<SupportIssueDto> {
+    if (!query.type && !query.id) throw new BadRequestException('Type or id is required');
+
+    const supportIssue = await this.supportIssueRepo.findOneBy({
+      userData: { id: userDataId },
+      type: query.type,
+      id: query.id,
+    });
+
+    if (!supportIssue) throw new NotFoundException('Support issue not found');
+
+    const messages = await this.messageRepo.findBy({ issue: { id: supportIssue.id } });
+    supportIssue.messages =
+      query.fromMessageId > 0
+        ? messages.filter((m) => m.id > query.fromMessageId)
+        : query.lastMessages > 0
+        ? messages.slice(-query.lastMessages)
+        : messages;
+
+    return SupportIssueDtoMapper.mapSupportIssue(supportIssue);
+  }
+
+  async getSupportIssueFile(userDataId: number, name: string): Promise<BlobContent> {
+    const allDocuments = await this.storageService.listUserFiles(userDataId);
+    const document = allDocuments.find((d) => d.name.includes(name));
+    if (!document) throw new NotFoundException('File not found');
+
+    return this.storageService.downloadFile(userDataId, document.type, document.name);
   }
```

Comment:
We are currently moving all support files to a different container, means we will need to change things a bit here.